console/console.go

```diff
@@ -99,26 +114,40 @@ func getConsolePluginCR(consolePort int, serviceName string, deploymentNamespace
 	}
 }
 
-func InitConsole(ctx context.Context, client client.Client, odfPort int, deploymentNamespace string) error {
-	deployment := appsv1.Deployment{}
+func InitConsole(ctx context.Context, client client.Client, scheme *runtime.Scheme, odfPort int, deploymentNamespace string) error {
+	mcoConsoleDeployment := appsv1.Deployment{}
 	if err := client.Get(context.TODO(), types.NamespacedName{
 		Name:      odfMulticlusterPluginName,
 		Namespace: deploymentNamespace,
-	}, &deployment); err != nil {
+	}, &mcoConsoleDeployment); err != nil {
+		return err
+	}
+
+	// Create core ODF multicluster console ConfigMap (nginx configuration)
+	mcoConsoleConfigMap := getNginxConfConfigMap(deploymentNamespace)
+	if _, err := controllerutil.CreateOrUpdate(ctx, client, &mcoConsoleConfigMap, func() error {
+		// Deployment deletion should delete corresponding ConfigMap as well
+		return controllerutil.SetControllerReference(&mcoConsoleDeployment, &mcoConsoleConfigMap, scheme)
+	}); err != nil {
 		return err
 	}
+
 	// Create core ODF multicluster console service
-	odfService := getService(odfMulticlusterPluginName, odfPort, deploymentNamespace)
-	if _, err := controllerutil.CreateOrUpdate(ctx, client, &odfService, func() error {
-		return nil
+	mcoConsoleService := getService(odfMulticlusterPluginName, odfPort, deploymentNamespace)
+	if _, err := controllerutil.CreateOrUpdate(ctx, client, &mcoConsoleService, func() error {
+		// Deployment deletion should delete corresponding Service as well
+		return controllerutil.SetControllerReference(&mcoConsoleDeployment, &mcoConsoleService, scheme)
 	}); err != nil {
 		return err
 	}
+
 	// Create core ODF multicluster plugin
-	odfConsolePlugin := getConsolePluginCR(odfPort, odfService.ObjectMeta.Name, deploymentNamespace)
-	if _, err := controllerutil.CreateOrUpdate(ctx, client, &odfConsolePlugin, func() error {
+	mcoConsolePlugin := getConsolePluginCR(odfPort, mcoConsoleService.ObjectMeta.Name, deploymentNamespace)
+	if _, err := controllerutil.CreateOrUpdate(ctx, client, &mcoConsolePlugin, func() error {
 		return nil
-	}); err != nil {
+	}); err != nil && !errors.IsAlreadyExists(err) {
+		// ConsolePlugin is a cluster-wide resource, and is not cleaned-up on operator un-install
```

Comment:
it is CreateOrUpdate only, is this will raise an already exists error?

Reply:
I did not test this... but seems to me:
Let;s say I installed MCO for the first time >> operator will create ConsolePlugin CR for me.
Now, I deleted MCO operator >> ConsolePlugin CR did not get deleted as a part of operator or NameSpace deletion.
Now, I install MCO again >> operator will create ConsolePlugin again, but this time it will already be existing as it was not automatically cleaned up at time of operator deletion... it that case it will return "alreadyExists" error, and here I am ignoring that error.

Reply:
ok! but I did this multiple times, I never faced this issue, I think it was because of createOrUpdate, anyway you are gonna this.

Reply:
actually it make sense... u r right "createOrUpdate" will handle this... "IsAlreadyExists" check is not needed...

Reply:
removed...